playwright/tests/shippingMethods.spec.ts

```diff
@@ -86,20 +86,39 @@ test("TC: SALEOR_33 Add weight rate to shipping method - with included zip codes
-                                                                                                                         page,
-                                                                                                                       }) => {
+test("TC: SALEOR_34 Delete a single shipping rate from the shipping zone details page @shipping-method @e2e", async ({
+                                                                                                                       page,
+                                                                                                                     }) => {
   const shippingMethodsPage = new ShippingMethodsPage(page);
   await shippingMethodsPage.gotoExistingShippingMethod(
     SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.id,
   );
   await expect(shippingMethodsPage.basePage.pageHeader).toBeVisible();
   const priceBasedRate = SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.rates.priceBasedRateToBeDeleted.name;
   await expect(shippingMethodsPage.priceBasedRatesSection).toContainText(priceBasedRate);
-  await shippingMethodsPage.clickDeleteShippingMethod();
+  await shippingMethodsPage.clickDeleteShippingRateFromTheList();
   await shippingMethodsPage.deleteShippingMethodDialog.clickDeleteButton();
   await shippingMethodsPage.basePage.expectSuccessBanner();
   await expect(shippingMethodsPage.priceBasedRatesSection).toContainText("No shipping rates found");
   await expect(shippingMethodsPage.priceBasedRatesSection).not.toContainText(priceBasedRate);
 });
 
+test("TC: SALEOR_35 Delete a single shipping rate from its details page @shipping-method @e2e", async ({
+                                                                                                         page,
+                                                                                                       }) => {
+  const shippingMethodsPage = new ShippingMethodsPage(page);
+  const shippingMethodId = SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.id;
+  const shippingRateId = SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.rates.weightBasedRateToBeDeleted.id;
+  const weightBasedRate = SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.rates.weightBasedRateToBeDeleted.name;
+
+  await shippingMethodsPage.gotoExistingShippingRate(shippingMethodId, shippingRateId);
+  await expect(shippingMethodsPage.basePage.pageHeader).toBeVisible();
+  await shippingMethodsPage.clickDeleteShippingRateButton();
+  await shippingMethodsPage.deleteShippingMethodDialog.clickDeleteButton();
+  await shippingMethodsPage.basePage.expectSuccessBanner();
+  await shippingMethodsPage.gotoExistingShippingMethod(
+    SHIPPING_METHODS.shippingMethodWithRatesToBeDeleted.id,
+  );
+  await expect(shippingMethodsPage.weightBasedRatesSection).toContainText("No shipping rates found");
+  await expect(shippingMethodsPage.weightBasedRatesSection).not.toContainText(weightBasedRate);
```

Comment:
We do not check integration between views so I would skip navigating to other views to confirm the action result. Check all you need on the screen that was loaded after deletion, do not go to other views to double-check results. Otherwise, this test would not have an end, there are so many views to be checked

Reply:
https://github.com/saleor/saleor-dashboard/pull/4625/commits/3f2bca7a34276fee0d0f6a299d359f77124979cb